### PR TITLE
Capture errors, localized messages to user, send to telemetry

### PR DESCRIPTION
**Pull request by dehru**
### What does this PR do?
This PR cleans up a [previous PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/2795) and incorporates that feedback.

This PR captures various errors, conditionally messages the user, and optionally sends them to telemetry as exceptions. 

### What issues does this PR fix or reference?
@W-8086238@

### Functionality Before
No exceptions sent to telemetry, inconsistent messages to the user, no i18n

### Functionality After
Some exceptions are available in app insights.  Emily has reviewed messaging to user.  I've incorporated feedback on which messaging options should be used for which types of messages.   Messages have been localized.

### CODE REVIEW

File: packages/salesforcedx-vscode-soql/src/commonUtils.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/commonUtils.ts:40]: The telemetry "problem_id" field showing in app insights looks like this.

app-insights-exceptions

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:143]: This rejected block has never thrown.  It's theoretical.  I even tried messing with CSP.  Still couldn't get it to hit.  But this would be catastrophic to our app if we can't communicate across the webview iframe....so i left it.  The error would look like SIMILAR to this in the channel and instruct the user to open an issue.

Screen Shot 2020-12-08 at 10 37 08 AM

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:193]: syntax-=unsupported

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:212]: An example of this message.  This would basically only happen if the soql-builder-ui and soql-extension versions weren't compatible.

unknown-message

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:220]: I'm not sure if both channel and poptarts are both necessary here.  But it sure gets the point across.

no-default-org


no-default-org-channel

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:233]: The information returned from the soql query runner api is very useful in debugging your soql statement issue.  @ekapner - once i found the useful information from the api, i gutted most of the message you carefully created.  All that is left is `We can't run the soql query."  Let me know if that's approprate.

run-query-error

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:253]: There will be conflicts with @dobladez 's [PR ](https://github.com/forcedotcom/salesforcedx-vscode/pull/2792) as these methods have been refactored.  But the message will likely remain the same and simply appear in the channel.

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:272]: After Fernando's Pr merge, i'll align this with his changes.  And include his check for `if (e.errorCode !== 'NOT_FOUND')`.  Otherwise, message will look like:

sobject-metadata-request

File: packages/salesforcedx-vscode-soql/src/messages/i18n.ja.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/messages/i18n.ja.ts:19]: @ekapner - Here's all the messages in a single place.

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts:91]: save-results-success

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts:63]: Again, i've never been able to trigger this theoretical exception with communicating across webview/iframe.

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts:140]: Likely, this error is thrown if soql-builder-ui and soql-extension versions are not functionally in sync.  Let's hope that never happens.

unknown-message

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts:159]: There must be some OS issue, permissions, or unknown, that would cause the file save issue.

save-results-failure

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (ekapner) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:233]: I'm not in love with that message but it does provide more pertinent details.

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts
Comment (ekapner) [reply packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts:91]: or We saved the results in the query-folder as:

**Review by ekapner — COMMENTED** [review]
One minor nit/suggested change to error message wording.

File: packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts
Comment (dehru) [reply packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts:91]: Updated.

File: packages/salesforcedx-vscode-soql/src/commonUtils.ts
Comment (jgellin-sf) [reply packages/salesforcedx-vscode-soql/src/commonUtils.ts:40]: nice

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts
Comment (jgellin-sf) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:134]: Nice consolidation

**Review by jgellin-sf — APPROVED** [review]
You've been tidying!

**Comment by dehru** [comment]
> You've been tidying!

@jgellin-sf - I spent much longer on this PR than intended.  And trying to trigger these errors.  And using the vsix to generate telemetry.  Emily has been patient with me too through 2 PRs!  

I'm ready to move onto something new. ;-)

**Review by dobladez — APPROVED** [review]
LGTM

**Comment by jgellin-sf** [comment]
QA is complete on this item.

@forcedotcom/pdt Please provide a code owner review when you can.

**Review by lcampos — APPROVED** [review]

**Pull request merged**
